src/main/java/com/springboot/star_wars_character_popularity/app/controller/VoteController.java

```diff
@@ -47,4 +54,37 @@ public ResponseEntity<String> deleteVote(@PathVariable("id") long id){
 
         return new ResponseEntity<String>("Vote deleted successfully !", HttpStatus.OK);
     }
+
+    public static String uploadDirectory = System.getProperty("user.dir")+"/uploads";
+
+    @RequestMapping("/voteView")
+    public ModelAndView UploadPage(Model model){
+
+        ModelAndView modelAndView = new ModelAndView();
+        modelAndView.setViewName("votes/vote-view");
+
+        return modelAndView;
+    }
+
+    @RequestMapping("/upload")
+    public ModelAndView upload(Model model, @RequestParam("image") MultipartFile[] files){
```

Comment:
use multipart form request
image + json data


move all code to service method